File: databinding/src/main/java/com/example/anvil/databinding/CheckBoxLayout.java
package com.example.anvil.databinding;

import android.content.Context;
import android.widget.CompoundButton;
import android.widget.LinearLayout;

import static trikita.anvil.DSL.*;
import trikita.anvil.RenderableView;

public class CheckBoxLayout extends RenderableView {

    boolean checked;

    public CheckBoxLayout(Context c) {
        super(c);
    }

    public void view() {
        linearLayout(() -> {
            size(FILL, FILL);
            orientation(LinearLayout.VERTICAL);
            padding(dip(20));

            checkBox(() -> {
                size(WRAP, WRAP);
                margin(0, dip(20));
                checked(checked);
                onCheckedChange((CompoundButton v, boolean isChecked) -> {
                    checked = isChecked;
                });
            });

            checkBox(() -> {
                size(WRAP, WRAP);
                margin(0, dip(20));
                checked(checked);
                onCheckedChange((CompoundButton v, boolean isChecked) -> {
                    checked = isChecked;
                });
            });
        });
    }
}

File: databinding-kotlin/src/main/kotlin/com/example/anvil/databinding/CheckBoxLayout.kt
package com.example.anvil.databinding

import android.content.Context
import android.widget.LinearLayout

import trikita.anvil.DSL.*
import trikita.anvil.RenderableView

class CheckBoxLayout(c: Context) : RenderableView(c) {

    var checked: Boolean = false

    override fun view() {
        linearLayout {
            size(FILL, FILL)
            orientation(LinearLayout.VERTICAL)
            padding(dip(20))

            checkBoxLayout()
            checkBoxLayout()
        }
    }

    fun checkBoxLayout() {
        checkBox {
            size(WRAP, WRAP)
            margin(0, dip(20))

            checked(checked)
            onCheckedChange { v, isChecked: Boolean -> checked = isChecked }
        }
    }
}
